src/frontends/sql/decoder_context_test.cc

```diff
@@ -166,4 +167,67 @@ INSTANTIATE_TEST_SUITE_P(DecodeMergeOpTest, DecodeMergeOpTest,
                          Combine(ValuesIn(kSampleInputVectors),
                                  ValuesIn(kSampleInputVectors)));
 
+class DecodeTagTransformTest
+    : public TestWithParam<
+          std::tuple<ir::Value, absl::string_view, std::vector<ir::Value>,
+                     std::vector<uint64_t>>> {
+ protected:
+  DecodeTagTransformTest() : ir_context_(), decoder_context_(ir_context_),
+  value_vec_(std::get<2>(GetParam())) {
+    // The id_sequence_vec is an id sequence that may have more IDs than we
+    // need for giving each value an ID. We will put a value_vec.size()
+    // prefix of it into input_id_vec_ so input_id_vec_ contains only those IDs
+    // associated with a value.
+    const std::vector<uint64_t> &id_sequence_vec = std::get<3>(GetParam());
+
+    input_id_vec_.reserve(value_vec_.size());
+    for (uint64_t idx = 0; idx < value_vec_.size(); ++idx) {
+      uint64_t id = id_sequence_vec.at(idx);
+      decoder_context_.RegisterValue(id, value_vec_.at(idx));
+      input_id_vec_.push_back(id);
+    }
+  }
+
+  IRContext ir_context_;
+  DecoderContext decoder_context_;
+  const std::vector<Value> &value_vec_;
```

Comment:
Instead of saving this in a field, have a helper that returns this as needed. e.g., 

```C++
const std::vector<Value>& GetInputValuesParam() const { return std::get<2>(GetParam()); }
```

Reply:
Got rid of the fixture.